src/mon/NVMeofGwMap.h

```diff
@@ -0,0 +1,92 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2023 IBM, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ */
+
+#ifndef MON_NVMEOFGWMAP_H_
+#define MON_NVMEOFGWMAP_H_
+#include <map>
+#include <iostream>
+#include "include/encoding.h"
+#include "include/utime.h"
+#include "common/Formatter.h"
+#include "common/ceph_releases.h"
+#include "common/version.h"
+#include "common/options.h"
+#include "common/Clock.h"
+#include "msg/Message.h"
+#include "common/ceph_time.h"
+#include "NVMeofGwTypes.h"
+
+using ceph::coarse_mono_clock;
+class Monitor;
+/*-------------------*/
+class NVMeofGwMap
+{
+public:
+    Monitor*                            mon           = NULL;
```

Comment:
Why does the map have a `Monitor*`? This seems very out-of-place.

Reply:
@gregsfortytwo, encoding of `NVMeofGwMap` distributed to the clients does not include Monitor reference, though agree this member looks odd, and could be refactored.

Reply:
The problem is that you just have to know, magically, which functions can be invoked on the client side and which can be invoked by the monitor.
In the rest of the Ceph code, manipulations which change map state for the future are performed by the monitor, using monitor code, and map functions are things which are available wherever the data structure happens to be available.

Reply:
@gregsfortytwo I think I might have mislead you. There are two two distinct classes with similar names here NVMeofGwMap, the one we're discussing and [MNVMeofGwMap](https://github.com/baum/ceph/blob/ceph-nvmeof-mon/src/messages/MNVMeofGwMap.h), distributed to clients and does not include Monitor reference. So I hope we're OK here. WDYT?

Reply:
@gregsfortytwo ^

Reply:
We talked about this in our meeting. Maps are really supposed to be the actual data structure that gets sent out over the wire, so the way this is encapsulated is not great. I'd certainly like to see it be Ceph-idiomatic but we might be able to continue as-is with good documentation about the architecture?